### ISSUE TITLE
Issue in training on another dataset

### ISSUE THREAD

**Issue by michaelz9436** [MuZhao2333/MolFLAE#2]
Hi, thanks for your work. I tried to train molflae on another dataset (with 2.1M mols, mean number of atom is 30, and size of atoms vocabulary is 11), I trained about 6 epoches, and the reconstructed molecule seems good but the pose (rotation) differs a lot from GT. Is this normal in your training process, that the rotation of reconstructed molecule will be inaccurate in the early epoches? Thanks for your help.

**Comment by MuZhao2333** [comment]
Hi, thanks for your interest in our work.
In our training process, the rotation is generally accurate. Could you please send me your loss curves and the original/reconstructed molecules for me to take a look? My email is chen-zt23@mails.tsinghua.edu.cn.
Additionally, you could try reducing the noise in the x component. Specifically, in `train_loop.py`, in the `encode` function, you can modify this part:
```python
upper = torch.log(torch.tensor(self.cfg['train']['kl_loss']['sigma2']**2, device=Zx_mu.device, dtype=Zx_mu.dtype))
raw = self.Wx_log_var(global_h).expand_as(Zx_mu)
Zx_log_var = torch.clamp(raw, max=upper)
```
Change `Zx_log_var` to be clamped to 0 (same as `Zh`), which might solve your problem:
Sorry for accidentally closing the issue earlier. I'm still playing with the buttons since this is my first time replying to an issue 😅. Feel free to continue contacting me via email.

**Comment by michaelz9436** [comment]
Thanks a lot for your reply. I tried to decrease the KL loss and now the reconstruction is good.
